Composer/packages/client/src/CreationFlow/DefineConversation/index.tsx

```diff
@@ -99,7 +99,10 @@ export const DefineConversation: React.FC<DefineConversationProps> = props => {
   };
 
   useEffect(() => {
-    const currentPath = Path.join(focusedStorageFolder.parent, focusedStorageFolder.name);
+    const currentPath =
+      Object.keys(focusedStorageFolder).length > 0
+        ? Path.join(focusedStorageFolder.parent, focusedStorageFolder.name)
+        : '';
```

Comment:
I'm still confused about why this needs to happen. When do we fetch the data to populate `focusedStorageFolder`? To me that seems like an issue here. We should fetch that before rendering the file selector.

Reply:
This line is a hot fix. Currenly, we have Storage and FocusedStorageFolder in global state. In Storage, we saved a path which will be used to update FocusedStorageFolder. The code is like 
useEffect(()=> update FocusedStorageFolder, [Storage]). I will refactor this part soon to remove this useEffect

The reason here I have to add a null props checking is: 
If creating a new bot, it is like 
Click New bot Button => fetch storage => finish updating storage => fetch FocusedStorageFolder => finish updating FocusedStorageFolder => user click Next => show folder info.
If opening a bot
Click open bot button => fetch storage => finish updating storage=>  Need to show Folder info now but we do not get FocuesdStoragerFolder yet, the UI start to render => app crush

Reply:
Can you fix this now?

Like maybe show a loading spinner if focused storage folder hasn't been set yet?

Reply:
Ping @liweitian. What is the plan here?

Reply:
We already have a spinner inside the fileSelector. I moved that part up to its parent component with checking on the  focusedStorageFolder.